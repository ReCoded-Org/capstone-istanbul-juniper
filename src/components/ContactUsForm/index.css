.contactUsFormContainer {
  background: #add4c5;
  display: flex;
  justify-content: center;
  padding-bottom: 1rem;
}

.contactUsForm__label {
  font-family: "Raleway";
  font-size: clamp(1rem, 24px, 3rem);
  color: white;
  text-align: left;
  margin-bottom: 0.6rem;
}

.contactUsForm__input,
.contactUsForm__textarea {
  font-family: "Raleway";
  font-size: clamp(1rem, 20px, 2.5rem);
  box-sizing: border-box;
  border: 2px solid #ccc;
  border-radius: 0.5rem;
  width: clamp(25rem, 17rem, 35rem);
  margin-bottom: 0.5rem;
}

.contactUsForm__textarea {
  height: clamp(10rem, 7rem, 15rem);
}

.contactUsForm__submit {
  margin-top: 0.5rem;
}

.contactUsForm__submitButton {
  border-radius: 10px;
  border: none;
  background: #f8ac4c;
  margin: auto;
  font-family: "Salsa", cursive;
  transition: all 0.5s;
  padding: 5px;
  font-size: clamp(0.75rem, 1.25rem, 1.5rem);
  cursor: pointer;
  color: white;
  width: clamp(6rem, 7rem, 8rem);
}

/* Form post submission message styling */
.contactUsForm__postSubmissionMessage {
  color: white;
  font-family: "Raleway";
  font-size: clamp(1.5rem, 30px, 3rem);
  text-align: center;
  box-sizing: border-box;
  background-color: #2a8c79;
  border-radius: 10px;
}
