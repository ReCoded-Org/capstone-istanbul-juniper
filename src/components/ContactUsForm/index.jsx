import React, { useState } from "react";
import { useTranslation } from "react-i18next";
import { Col, Row } from "antd";
import { RiSendPlaneFill } from "react-icons/ri";
import "./index.css";

// Converts object to proper form data(string)
const encodeFormData = (data) => {
  return Object.keys(data)
    .map((key) => encodeURIComponent(key) + "=" + encodeURIComponent(data[key]))
    .join("&");
};

const ContactUsForm = () => {
  const [t] = useTranslation();
  const [formValue, setFormValue] = useState({
    name: "",
    email: "",
    message: "",
  });

  // The change in isSubmitted state triggers post submission message
  const [isSubmitted, setIsSubmitted] = useState(false);

  const handleChange = (e) => setFormValue({ [e.target.name]: e.target.value });

  // application/x-www-form-urlencoded stands for POST data at HTTP.
  // body represents HTTP message and send data to server in a query string format.
  // encode is used for passing the string to the body
  // setTimeOut stands for showing post submission message about a time after submit
  const handleSubmit = (e) => {
    fetch("/", {
      method: "POST",
      "data-netlify": "true",
      headers: { "Content-Type": "application/x-www-form-urlencoded" },
      body: encodeFormData({ "form-name": "contact", ...formValue }),
    }).then(() => {
      setTimeout(() => setIsSubmitted(false), 2000);
      setIsSubmitted(true);
      setFormValue({
        name: "",
        email: "",
        message: "",
      }).catch((error) => alert(error));

      e.preventDefault();
    });
  };

  return (
    <div className="contactUsFormContainer">
      <form onSubmit={handleSubmit}>
        <Col>
          <div
            className="contactUsForm__postSubmissionMessage"
            style={{ visibility: isSubmitted ? "visible" : "hidden" }}
          >
            <strong>{t("contact.contactUsForm.postSubmissionMessage")}</strong>
          </div>
          <Col>
            <label className="contactUsForm__label">
              {t("contact.contactUsForm.name")}
            </label>
          </Col>
          <Col>
            <input
              type="text"
              name="name"
              value={formValue.name}
              onChange={handleChange}
              className="contactUsForm__input"
              required
            />
          </Col>
        </Col>
        <Col>
          <Col>
            <label className="contactUsForm__label">
              {t("contact.contactUsForm.email")}
            </label>
          </Col>
          <Col>
            <input
              type="email"
              name="email"
              value={formValue.email}
              onChange={handleChange}
              className="contactUsForm__input"
              required
            />
          </Col>
        </Col>
        <Col>
          <Col>
            <label className="contactUsForm__label">
              {t("contact.contactUsForm.message")}
            </label>
          </Col>
          <Col>
            <textarea
              name="message"
              value={formValue.message}
              onChange={handleChange}
              className="contactUsForm__textarea"
              required
            />
          </Col>
        </Col>
        <Row>
          <div className="contactUsForm__submit">
            <button type="submit" className="contactUsForm__submitButton">
              {t("contact.contactUsForm.send")}
              <RiSendPlaneFill />
            </button>
          </div>
        </Row>
      </form>
    </div>
  );
};

export default ContactUsForm;
