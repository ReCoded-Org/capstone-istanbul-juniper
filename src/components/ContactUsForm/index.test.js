import React from "react";
import ContactUsForm from "./index.jsx";
import renderer from "react-test-renderer";

it("ContactUsForm snapshot test", () => {
  const tree = renderer.create(<ContactUsForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
