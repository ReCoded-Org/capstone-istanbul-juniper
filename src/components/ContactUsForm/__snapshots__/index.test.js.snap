// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactUsForm snapshot test 1`] = `
<div
  className="contactUsFormContainer"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="ant-col"
      style={Object {}}
    >
      <div
        className="contactUsForm__postSubmissionMessage"
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        <strong>
          contact.contactUsForm.postSubmissionMessage
        </strong>
      </div>
      <div
        className="ant-col"
        style={Object {}}
      >
        <label
          className="contactUsForm__label"
        >
          contact.contactUsForm.name
        </label>
      </div>
      <div
        className="ant-col"
        style={Object {}}
      >
        <input
          className="contactUsForm__input"
          name="name"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="ant-col"
      style={Object {}}
    >
      <div
        className="ant-col"
        style={Object {}}
      >
        <label
          className="contactUsForm__label"
        >
          contact.contactUsForm.email
        </label>
      </div>
      <div
        className="ant-col"
        style={Object {}}
      >
        <input
          className="contactUsForm__input"
          name="email"
          onChange={[Function]}
          required={true}
          type="email"
          value=""
        />
      </div>
    </div>
    <div
      className="ant-col"
      style={Object {}}
    >
      <div
        className="ant-col"
        style={Object {}}
      >
        <label
          className="contactUsForm__label"
        >
          contact.contactUsForm.message
        </label>
      </div>
      <div
        className="ant-col"
        style={Object {}}
      >
        <textarea
          className="contactUsForm__textarea"
          name="message"
          onChange={[Function]}
          required={true}
          value=""
        />
      </div>
    </div>
    <div
      className="ant-row"
      style={Object {}}
    >
      <div
        className="contactUsForm__submit"
      >
        <button
          className="contactUsForm__submitButton"
          type="submit"
        >
          contact.contactUsForm.send
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M1.946 9.315c-.522-.174-.527-.455.01-.634l19.087-6.362c.529-.176.832.12.684.638l-5.454 19.086c-.15.529-.455.547-.679.045L12 14l6-8-8 6-8.054-2.685z"
              />
            </g>
          </svg>
        </button>
      </div>
    </div>
  </form>
</div>
`;
